src/lobsterpy/featurize/core.py

```diff
@@ -1146,3 +1132,462 @@ def get_fingerprint_df(
+
+        :param bwdf: BWDF data as a dictionary
+
+        Returns:
+            Normalized BWDF data as a dictionary
+        """
+        for bwdf_label, bwdf_value in bwdf.items():
+            if bwdf_label not in ("centers", "edges", "bin_width"):
+                if self.normalization == "area":
+                    total_area = np.sum(np.abs(bwdf_value["icoxx_binned"]) * self.bin_width)
+                    bwdf[bwdf_label]["icoxx_binned"] = np.nan_to_num(bwdf_value["icoxx_binned"] / total_area)
+                elif self.normalization == "formula_units":
+                    formula_units = self.structure.composition.get_reduced_formula_and_factor()[-1]
+                    bwdf[bwdf_label]["icoxx_binned"] = np.nan_to_num(bwdf_value["icoxx_binned"] / formula_units)
+                elif self.normalization == "counts":
+                    bwdf[bwdf_label]["icoxx_binned"] = np.nan_to_num(
+                        bwdf_value["icoxx_binned"] / bwdf_value["icoxx_counts"]
+                    )
+
+        return bwdf
+
+    def get_df(self, ids: str | None = None) -> pd.DataFrame:
+        """Return a pandas dataframe with computed BWDF features as columns.
+
+        Args:
+            ids: set index name in the pandas dataframe. Default is None.
+
+        Returns:
+            A pandas dataframe object with BWDF features
+        """
+        bwdf = self.calc_bwdf()
+        column_names = self._get_features_col_names(bwdf=bwdf)
+        if ids:
+            df = pd.DataFrame(index=[ids], columns=column_names)
+        else:
+            ids = Path(self.path_to_icoxxlist).parent.name
+            df = pd.DataFrame(index=[ids], columns=column_names)
+
+        for icoxx_weight, col in zip(bwdf["summed"]["icoxx_binned"], column_names):
+            df.loc[ids, col] = icoxx_weight
+
+        return df
+
+    def get_site_df(self, site_index: int, ids: str | None = None) -> pd.DataFrame:
+        """Return a pandas dataframe with computed BWDF features for a site as columns.
+
+        Args:
+            site_index: index of the site in a structure for which BWDF needs to be computed
+            ids: set index name in the pandas dataframe. Default is None.
+
+        Returns:
+            A pandas dataframe object with BWDF features for a site
+        """
+        site_bwdf = self.calc_site_bwdf(site_index=site_index)
+        column_names = [f"{feat_name}_site_{site_index}" for feat_name in self._get_features_col_names(bwdf=site_bwdf)]
+        if ids:
+            df = pd.DataFrame(index=[ids], columns=column_names)
+        else:
+            ids = Path(self.path_to_icoxxlist).parent.name
+            df = pd.DataFrame(index=[ids], columns=column_names)
+
+        for icoxx_weight, col in zip(site_bwdf[f"{site_index}"]["icoxx_binned"], column_names):
+            df.loc[ids, col] = icoxx_weight
+
+        return df
+
+    def get_stats_df(self, ids: str | None = None) -> pd.DataFrame:
```

Comment:
Hi @naik-aakash another BWDF featurization that avoids many zero entries could be (a) a list of distances ordered by BWDF value, descending or (b) a list of BWDF values ordered by distances, ascending. If you are interested, i'm happy to add that.

Reply:
I think this sounds like a very good idea.

Reply:
Thanks @kaueltzen for this suggestion. Could be nice to also test this out. I have now pushed the changes to repo, you can add it to this branch